pkg/securitycontextconstraints/sccadmission/admission.go

```diff
@@ -227,12 +242,59 @@ loop:
 			break loop
 		default:
 			klog.V(5).Infof("pod %s (generate: %s) validated against provider %s, but required mutation, skipping", pod.Name, pod.GenerateName, sccName)
+			failed[sccName] = fmt.Sprintf("failed final validation after mutating admission")
+		}
+	}
+
+	// add audit annotations
+	if specMutationAllowed {
+		a.AddAnnotation("securityserviceconstraints.admission.openshift.io/denied", strings.Join(denied, ","))
+		for sccName, reason := range failed {
+			a.AddAnnotation(fmt.Sprintf("securitycontextconstraints.admission.openshift.io/too-restrictive-%s", sccName), reason)
+		}
+
+		if allowingProvider != nil && nextProvider != nil {
+			chosen, _ := c.sccLister.Get(allowingProvider.GetSCCName())
```

Comment:
why dropping the error? guess this is baked by informer, so an uninitialized informer will get 404?

Reply:
fixed